Quadriga/src/main/java/edu/asu/spring/quadriga/accesschecks/impl/ProjectSecurityChecker.java

```diff
@@ -190,34 +191,37 @@ public boolean checkProjectAccess(String userName, String projectId)
         chkAccess = this.isProjectOwner(userName, projectId);
 
         if (!chkAccess) {
-            chkAccess = this.isUserCollaboratorOnProject(userName, projectId,
-                    RoleNames.ROLE_QUADRIGA_ADMIN);
+            chkAccess = this.isUserCollaboratorOnProject(userName, projectId, RoleNames.ROLE_QUADRIGA_ADMIN);
         }
         return chkAccess;
     }
 
-    /**
-     * This method checks if the user has the editor role to
-     * this Project
-     * 
-     * @param userName
-     * @return boolean - TRUE if the user is editor on the project; otherwise false
-     * @throws QuadrigaStorageException
-     * @author kiranbatna
-     */
     @Override
     @Transactional
-    public boolean isEditor(String userName,
-            String projectId) throws QuadrigaStorageException {
-        // check if the user is project editor
-        return accessManager
-                .isUserEditorOfProject(userName, projectId) > 0;
+    public boolean isUnixnameInUse(String unixName, String projectId) throws QuadrigaStorageException {
+        return accessManager.getProjectIdByUnixName(unixName) != null;
     }
 
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.spring.quadriga.accesschecks.IProjectSecurityChecker#
+     * getCollaboratorRoles(java.lang.String, java.lang.String)
+     */
     @Override
-    @Transactional
-    public boolean isUnixnameInUse(String unixName, String projectId)
-            throws QuadrigaStorageException {
-       return accessManager.getProjectIdByUnixName(unixName) != null;
+    public Map<String, Boolean> getCollaboratorRoles(String userName, String projectId) {
+        Map<String, Boolean> collaboratorRolesMap = new HashMap<String, Boolean>();
+        collaboratorRolesMap.put(RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN, false);
+        collaboratorRolesMap.put(RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR, false);
+        collaboratorRolesMap.put(RoleNames.ROLE_PROJ_COLLABORATOR_EDITOR, false);
+
+        List<String> userDBRoles = accessManager.getProjectCollaboratorRoles(userName, projectId);
+
+        for (int i = 0; i < userDBRoles.size(); i++) {
+            collaboratorRolesMap.put(
+                    roleManager.getQuadrigaRoleByDbId(IQuadrigaRoleManager.PROJECT_ROLES, userDBRoles.get(i)).getId(),
+                    true);
+        }
+        return collaboratorRolesMap;
```

Comment:
When I return a list with all roles user has,  to check whether the user is editor, contributor etc I need to iterate through the list everytime to check, whereas with map it's O(1) acess everytime.